src/components/editMeasure/testCases/components/testCaseLanding/qiCore/TestCaseList.tsx

```diff
@@ -372,16 +372,25 @@ const TestCaseList = (props: TestCaseListProps) => {
     }
   };
 
+  const [selectedTestCaseIds, setSelectedTestCaseIds] = useState<string[]>();
+  useEffect(() => {
+    const testCaseIds: string[] = [];
+    if (selectedTestCases) {
+      selectedTestCases.forEach((testCase) => {
+        testCaseIds.push(testCase.id);
+      });
+    }
+    setSelectedTestCaseIds(testCaseIds);
+  }, [selectedTestCases]);
```

Comment:
I think this can be done without useEfect as well
```suggestion
  const selectedTestCaseIds = selectedTestCases?.map((testCase) => testCase.id);
```

Reply:
I don't understand how this is fixing the issue of "unable to export single test case"

Reply:
@adongare Thanks so much Ashok, it does work without the useEffect! @RohitKandimalla It was due to the selected testcases ids were undefined. I just used what Ashok suggested, it's working as well, yet better!

Reply:
So why is setSelectedTestCases.id undefinded ? Shouldn't it contain an ID ? why is it not available only when we are trying to export 1 test case ?

Reply:
@RohitKandimalla TestCase does have id, but selectedTestCases was set in TestCaseTable.tsx.

Reply:
Yeah so shouldn't we make sure Id are added to selectedTestCases in TestCaseTable.tsx ?

Reply:
@RohitKandimalla TestCaseTable set the selected test cases with their ids properly. I think the rerender happens with the useEffect in TestCaseTable when selectedTestCases are set, since I removed the useEffect in TestCaseList, there shouldn't be any extra rerenders?

Reply:
I am not saying we are creating re-renderes, I am asking if we relying on those re-renders. I will debug once, but the thing i am curious of is, if testCaseTable is setting theri ids properlty then why are are they undefined ?

Reply:
So the underlying issue is that selectedTestCases is looped with a prior condition selectedTestCases.size > 0 
but the thing is selectedTestCases is a state variable of type "any" hence our compiler never complained, but essentailly selectedTestCases should be a list and that when changed will throw compiler errors. Since selectedTestCases.size is returned as undefined the else block is executed, here the else is used to export all testCases inside a measure. So the resolution is to use appropriate size check, and I would say we should be using lodash _,size(selectedTestCases). In addition to avoid future compliation errors we should declare selectedTestCases as a TestCase List instead of any

Reply:
@RohitKandimalla thanks a lot Rohit, just made the change.